### PR TITLE
Tests

### PR DESCRIPTION
**Pull request by ericgibb**
- adds initial support for Detox E2E testing
- adds more jest snapshot tests

### CODE REVIEW

File: src/components/Button.js [hunk 1]
```diff
@@ -121,6 +123,7 @@ export default class Button extends Component {
 
     return (
       <TouchableOpacity
+        testID={this.props.testID}
```
Comment (zachgibb) [inline src/components/Button.js]: will `testID` be a necessary prop for every component?

File: src/components/GradientButton.test.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+import React from 'react';
+import GradientButton from './GradientButton';
+
+import renderer from 'react-test-renderer';
+
+// const onPressEvent = jest.fn();
+
+test('renders correctly', () => {
+  // const tree = renderer.create(<GradientButton />).toJSON();
+  const tree = true;
```
Comment (zachgibb) [inline src/components/GradientButton.test.js]: just to confirm, we are having these fail intentionally?

File: src/components/Button.js (hunk 1 above)
Comment (ericgibb) [reply src/components/Button.js]: yes, every custom component we want testable via an ID. detox adds an additional param (`accessibility identifier` I think?) and queries the actual primitive by that id. if we don't want to test for it, we won't need it.

File: src/components/GradientButton.test.js (hunk 2 above)
Comment (ericgibb) [reply src/components/GradientButton.test.js]: yeah... to force us to clean them up and test them. should they just be removed and picked up in code coverage tests?

Comment (zachgibb) [reply src/components/GradientButton.test.js]: i'm largely ambivalent about that. with these here, they alert us to the fact that we need to fix / build out these tests. 

fine for now, i think

Comment (ericgibb) [reply src/components/GradientButton.test.js]: 👍

File: src/components/Button.js (hunk 1 above)
Comment (zachgibb) [reply src/components/Button.js]: ok. will it need to be unique across the entire app for detox to work? or per-screen unique?

Comment (zachgibb) [reply src/components/Button.js]: adding a unique, stable id to every element i could forsee getting quite cumbersome

Comment (ericgibb) [reply src/components/Button.js]: it's passed in as a prop from wherever it's added. that's a good point though - but I don't think that should be an issue. if we're composing the same view multiple times in a single screen and wanting to check them independently, then yes maybe our tests will be confused? we can search by id or by text content though, so in those cases we'd have to modify our tests or update the id intelligently.

do you see a bigger issue with this, though?

Comment (zachgibb) [reply src/components/Button.js]: as discussed in-person: the uniqueness for these ids would be per-screen, which i don't see to be an issue. 👍

Comment (zachgibb) [reply src/components/Button.js]: (and even if it would be app-wide, it's technically possible, just annoying)

**Pull request merged**